llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -300,13 +300,13 @@ void WindowScheduler::generateTripleMBB() {
   // are updated accordingly.
   for (size_t Cnt = 1; Cnt < DuplicateNum; ++Cnt) {
     for (auto *MI : OriMIs) {
-      if (MI->isPHI() || MI->isDebugInstr() ||
+      if (MI->isPHI() || MI->isMetaInstruction() ||
           (MI->isTerminator() && Cnt < DuplicateNum - 1))
         continue;
       auto *NewMI = MF->CloneMachineInstr(MI);
       DenseMap<Register, Register> NewDefs;
       // New defines are updated.
-      for (auto MO : NewMI->defs())
+      for (auto MO : NewMI->all_defs())
```

Comment:
All these type of changes should have tests to go with them

Reply:
Okay, we have added test cases for dead def, implicit-def, and meta instruction.